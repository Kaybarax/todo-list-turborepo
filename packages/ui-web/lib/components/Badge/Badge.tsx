import React from 'react';
import { Slot } from '@radix-ui/react-slot';
import { cva, type VariantProps } from 'class-variance-authority';
import { cn } from '../../utils';

const badgeVariants = cva(
  'inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2',
  {
    variants: {
      variant: {
        default:
          'border-transparent bg-primary text-primary-foreground hover:bg-primary/80',
        secondary:
          'border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80',
        destructive:
          'border-transparent bg-destructive text-destructive-foreground hover:bg-destructive/80',
        outline: 'text-foreground border-border',
        success: 
          'border-transparent bg-green-500 text-white hover:bg-green-500/80',
        warning:
          'border-transparent bg-yellow-500 text-white hover:bg-yellow-500/80',
        info:
          'border-transparent bg-blue-500 text-white hover:bg-blue-500/80',
        ghost:
          'border-transparent bg-transparent text-foreground hover:bg-accent',
      },
      size: {
        default: 'px-2.5 py-0.5 text-xs',
        sm: 'px-2 py-0.5 text-xs',
        lg: 'px-3 py-1 text-sm',
      },
      shape: {
        default: 'rounded-full',
        square: 'rounded-md',
        pill: 'rounded-full',
      },
    },
    defaultVariants: {
      variant: 'default',
      size: 'default',
      shape: 'default',
    },
  }
);

export interface BadgeProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof badgeVariants> {
  asChild?: boolean;
  icon?: React.ReactNode;
  closable?: boolean;
  onClose?: () => void;
  dot?: boolean;
}

const Badge = React.forwardRef<HTMLDivElement, BadgeProps>(
  ({ 
    className, 
    variant, 
    size,
    shape,
    asChild = false,
    icon, 
    closable = false,
    onClose,
    dot = false,
    children, 
    ...props 
  }, ref) => {

    
    const handleClose = (e: React.MouseEvent) => {
      e.stopPropagation();
      onClose?.();
    };

    if (asChild) {
      return (
        <Slot 
          className={cn(badgeVariants({ variant, size, shape }), className)} 
          ref={ref}
          {...props}
        >
          {children}
        </Slot>
      );
    }

    return (
      <div 
        className={cn(badgeVariants({ variant, size, shape }), className)} 
        ref={ref}
        {...props}
      >
        {dot && (
          <span 
            className={cn(
              'mr-1.5 h-2 w-2 rounded-full',
              variant === 'outline' ? 'bg-current' : 'bg-current opacity-75'
            )}
            aria-hidden="true"
          />
        )}
        {!dot && icon && (
          <span className="mr-1.5 flex items-center" aria-hidden="true">
            {icon}
          </span>
        )}
        <span className="truncate">{children}</span>
        {closable && (
          <button
            type="button"
            className={cn(
              'ml-1.5 flex items-center justify-center rounded-full hover:bg-black/10 focus:outline-none focus:ring-1 focus:ring-current',
              'h-4 w-4 transition-colors'
            )}
            onClick={handleClose}
            aria-label="Remove badge"
          >
            <svg
              className="h-3 w-3"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              aria-hidden="true"
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                d="M6 18L18 6M6 6l12 12"
              />
            </svg>
          </button>
        )}
      </div>
    );
  }
);

Badge.displayName = 'Badge';

export { Badge, badgeVariants };