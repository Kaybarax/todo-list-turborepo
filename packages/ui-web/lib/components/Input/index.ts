export { Input, Label, inputVariants } from './Input';
export type { InputProps } from './Input';
