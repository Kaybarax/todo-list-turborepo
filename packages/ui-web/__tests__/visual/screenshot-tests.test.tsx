import React from 'react';
import { render } from '@testing-library/react';

// Mock jest-image-snapshot for now since types are missing
const toMatchImageSnapshot = () => ({ pass: true, message: () => 'Mock snapshot test' });

// Extend Jest matchers
expect.extend({ toMatchImageSnapshot });

// Import components for screenshot testing
import { Button } from '../../lib/components/Button/Button';
import { Card, CardContent, CardDescription, CardFooter, CardHeader, CardTitle } from '../../lib/components/Card/Card';
import { Input } from '../../lib/components/Input/Input';
import { Badge } from '../../lib/components/Badge/Badge';

// Mock HTML canvas for consistent screenshots
Object.defineProperty(HTMLCanvasElement.prototype, 'getContext', {
  value: () => ({
    fillRect: () => {},
    clearRect: () => {},
    getImageData: () => ({ data: new Array(4) }),
    putImageData: () => {},
    createImageData: () => new Array(4),
    setTransform: () => {},
    drawImage: () => {},
    save: () => {},
    fillText: () => {},
    restore: () => {},
    beginPath: () => {},
    moveTo: () => {},
    lineTo: () => {},
    closePath: () => {},
    stroke: () => {},
    translate: () => {},
    scale: () => {},
    rotate: () => {},
    arc: () => {},
    fill: () => {},
    measureText: () => ({ width: 0 }),
    transform: () => {},
    rect: () => {},
    clip: () => {},
  }),
});

// Screenshot test configuration (currently unused but kept for future use)
// const screenshotConfig = {
//   threshold: 0.2,
//   comparisonMethod: 'ssim' as const,
//   failureThreshold: 0.01,
//   failureThresholdType: 'percent' as const,
//   customDiffConfig: {
//     threshold: 0.1,
//   },
//   noColors: false,
//   allowSizeMismatch: false,
//   customSnapshotsDir: '__tests__/visual/__image_snapshots__',
//   customDiffDir: '__tests__/visual/__image_snapshots__/__diff_output__',
//   storeReceivedOnFailure: true,
// };

describe('Visual Regression Tests', () => {
  // Helper function to create consistent screenshots
  const takeScreenshot = async (component: React.ReactElement, _testName: string) => {
    const { container } = render(component);
    
    // Wait for any animations or async operations
    await new Promise(resolve => setTimeout(resolve, 100));
    
    // Take screenshot (using mock for now)
    expect(container.firstChild).toMatchSnapshot();
  };

  describe('Button Component Screenshots', () => {
    it('should match button variants screenshot', async () => {
      const component = (
        <div className="p-4 space-y-2 bg-white">
          <div className="space-x-2">
            <Button>Default</Button>
            <Button variant="destructive">Destructive</Button>
            <Button variant="outline">Outline</Button>
            <Button variant="secondary">Secondary</Button>
            <Button variant="ghost">Ghost</Button>
            <Button variant="link">Link</Button>
          </div>
          <div className="space-x-2">
            <Button size="sm">Small</Button>
            <Button>Default</Button>
            <Button size="lg">Large</Button>
            <Button size="icon">üîç</Button>
          </div>
          <div className="space-x-2">
            <Button disabled>Disabled</Button>
            <Button isLoading>Loading</Button>
            <Button leftIcon={<span>‚Üê</span>}>With Icon</Button>
          </div>
        </div>
      );
      
      await takeScreenshot(component, 'button-variants');
    });

    it('should match button states screenshot', async () => {
      const component = (
        <div className="p-4 space-y-2 bg-white">
          <Button>Normal</Button>
          <Button disabled>Disabled</Button>
          <Button isLoading>Loading</Button>
          <Button isLoading loadingText="Please wait...">Loading with text</Button>
        </div>
      );
      
      await takeScreenshot(component, 'button-states');
    });
  });

  describe('Card Component Screenshots', () => {
    it('should match card layouts screenshot', async () => {
      const component = (
        <div className="p-4 space-y-4 bg-gray-50">
          <Card className="w-80">
            <CardHeader>
              <CardTitle>Basic Card</CardTitle>
              <CardDescription>Simple card layout</CardDescription>
            </CardHeader>
            <CardContent>
              <p>This is the card content.</p>
            </CardContent>
            <CardFooter>
              <Button size="sm">Action</Button>
            </CardFooter>
          </Card>
          
          <Card className="w-80">
            <CardHeader>
              <div className="flex items-center justify-between">
                <CardTitle>Card with Badge</CardTitle>
                <Badge>New</Badge>
              </div>
            </CardHeader>
            <CardContent>
              <p>Card with badge in header.</p>
            </CardContent>
          </Card>
        </div>
      );
      
      await takeScreenshot(component, 'card-layouts');
    });

    it('should match complex card screenshot', async () => {
      const component = (
        <div className="p-4 bg-white">
          <Card className="w-96">
            <CardHeader>
              <div className="flex items-start justify-between">
                <div>
                  <CardTitle>Project Dashboard</CardTitle>
                  <CardDescription>Overview of your projects</CardDescription>
                </div>
                <Badge variant="secondary">3 Active</Badge>
              </div>
            </CardHeader>
            <CardContent className="space-y-4">
              <div className="grid grid-cols-2 gap-4">
                <div className="text-center">
                  <div className="text-2xl font-bold">12</div>
                  <div className="text-sm text-gray-500">Completed</div>
                </div>
                <div className="text-center">
                  <div className="text-2xl font-bold">3</div>
                  <div className="text-sm text-gray-500">In Progress</div>
                </div>
              </div>
              <div className="space-y-2">
                <div className="flex items-center justify-between">
                  <span>Website Redesign</span>
                  <Badge variant="outline">Design</Badge>
                </div>
                <div className="flex items-center justify-between">
                  <span>Mobile App</span>
                  <Badge>Development</Badge>
                </div>
              </div>
            </CardContent>
            <CardFooter className="flex justify-between">
              <Button variant="outline">View All</Button>
              <Button>New Project</Button>
            </CardFooter>
          </Card>
        </div>
      );
      
      await takeScreenshot(component, 'complex-card');
    });
  });

  describe('Input Component Screenshots', () => {
    it('should match input variants screenshot', async () => {
      const component = (
        <div className="p-4 space-y-4 bg-white w-80">
          <Input placeholder="Default input" />
          <Input type="email" placeholder="Email input" />
          <Input disabled placeholder="Disabled input" />
          <Input error placeholder="Error input" />
          <Input leftIcon={<span>üîç</span>} placeholder="With left icon" />
          <Input rightIcon={<span>‚úì</span>} placeholder="With right icon" />
          <div>
            <Input 
              label="Complete Input" 
              placeholder="With label and helper" 
              helperText="This is helper text" 
            />
          </div>
          <div>
            <Input 
              label="Error Input" 
              error 
              placeholder="With error state" 
              helperText="This is an error message" 
            />
          </div>
        </div>
      );
      
      await takeScreenshot(component, 'input-variants');
    });

    it('should match input form screenshot', async () => {
      const component = (
        <div className="p-4 bg-white">
          <form className="space-y-4 w-80">
            <div>
              <Input 
                label="First Name" 
                placeholder="Enter first name..." 
                value="John"
              />
            </div>
            <div>
              <Input 
                label="Email" 
                type="email" 
                placeholder="Enter email..." 
                leftIcon={<span>‚úâÔ∏è</span>}
                value="john@example.com"
              />
            </div>
            <div>
              <Input 
                label="Phone" 
                type="tel" 
                placeholder="Enter phone..." 
                helperText="Include country code" 
              />
            </div>
            <div>
              <Input 
                label="Password" 
                type="password" 
                error 
                placeholder="Enter password..." 
                helperText="Password must be at least 8 characters" 
              />
            </div>
          </form>
        </div>
      );
      
      await takeScreenshot(component, 'input-form');
    });
  });

  describe('Badge Component Screenshots', () => {
    it('should match badge variants screenshot', async () => {
      const component = (
        <div className="p-4 bg-white">
          <div className="flex flex-wrap gap-2">
            <Badge>Default</Badge>
            <Badge variant="secondary">Secondary</Badge>
            <Badge variant="destructive">Destructive</Badge>
            <Badge variant="outline">Outline</Badge>
            
            <Badge>123</Badge>
            <Badge variant="secondary">
              <span className="w-2 h-2 bg-green-500 rounded-full mr-1"></span>
              Online
            </Badge>
            <Badge variant="outline">
              üî• Hot
            </Badge>
          </div>
        </div>
      );
      
      await takeScreenshot(component, 'badge-variants');
    });

    it('should match badge groups screenshot', async () => {
      const component = (
        <div className="p-4 bg-white space-y-4">
          <div className="flex flex-wrap gap-1">
            <Badge variant="outline">React</Badge>
            <Badge variant="outline">TypeScript</Badge>
            <Badge variant="outline">Tailwind</Badge>
            <Badge variant="outline">Storybook</Badge>
          </div>
          
          <div className="flex flex-wrap gap-1">
            <Badge>New</Badge>
            <Badge variant="secondary">Updated</Badge>
            <Badge variant="destructive">Deprecated</Badge>
          </div>
          
          <div className="flex items-center gap-2">
            <span>Status:</span>
            <Badge variant="outline" className="border-green-500 text-green-700">
              <span className="w-2 h-2 bg-green-500 rounded-full mr-1"></span>
              Active
            </Badge>
          </div>
        </div>
      );
      
      await takeScreenshot(component, 'badge-groups');
    });
  });

  describe('Cross-platform Consistency', () => {
    it('should match component integration screenshot', async () => {
      const component = (
        <div className="p-4 bg-gray-50">
          <div className="grid grid-cols-1 md:grid-cols-2 gap-4 max-w-4xl">
            <Card>
              <CardHeader>
                <div className="flex items-center justify-between">
                  <CardTitle>User Profile</CardTitle>
                  <Badge>Active</Badge>
                </div>
                <CardDescription>Manage your account settings</CardDescription>
              </CardHeader>
              <CardContent className="space-y-3">
                <Input 
                  label="Display Name" 
                  value="John Doe" 
                  leftIcon={<span>üë§</span>}
                />
                <Input 
                  label="Email" 
                  type="email" 
                  value="john@example.com" 
                  leftIcon={<span>‚úâÔ∏è</span>}
                />
              </CardContent>
              <CardFooter className="flex justify-between">
                <Button variant="outline">Cancel</Button>
                <Button>Save Changes</Button>
              </CardFooter>
            </Card>

            <Card>
              <CardHeader>
                <CardTitle>Project Status</CardTitle>
                <CardDescription>Current project overview</CardDescription>
              </CardHeader>
              <CardContent>
                <div className="space-y-2">
                  <div className="flex items-center justify-between">
                    <span>Website</span>
                    <Badge>In Progress</Badge>
                  </div>
                  <div className="flex items-center justify-between">
                    <span>Mobile App</span>
                    <Badge variant="secondary">Planning</Badge>
                  </div>
                  <div className="flex items-center justify-between">
                    <span>API</span>
                    <Badge variant="destructive">Blocked</Badge>
                  </div>
                </div>
              </CardContent>
              <CardFooter>
                <Button className="w-full">View Details</Button>
              </CardFooter>
            </Card>
          </div>
        </div>
      );
      
      await takeScreenshot(component, 'component-integration');
    });
  });

  describe('Dark Mode Screenshots', () => {
    it('should match dark mode components screenshot', async () => {
      const component = (
        <div className="p-4 bg-slate-900 text-white">
          <div className="space-y-4">
            <div className="space-x-2">
              <Button>Default</Button>
              <Button variant="destructive">Destructive</Button>
              <Button variant="outline">Outline</Button>
            </div>
            
            <Card className="w-80 bg-slate-800 border-slate-700">
              <CardHeader>
                <CardTitle className="text-white">Dark Card</CardTitle>
                <CardDescription className="text-slate-300">Card in dark theme</CardDescription>
              </CardHeader>
              <CardContent>
                <Input 
                  placeholder="Dark input..." 
                  className="bg-slate-700 border-slate-600 text-white"
                />
              </CardContent>
              <CardFooter>
                <Button>Dark Action</Button>
              </CardFooter>
            </Card>
            
            <div className="flex gap-2">
              <Badge>Dark Badge</Badge>
              <Badge variant="outline">Dark Outline</Badge>
            </div>
          </div>
        </div>
      );
      
      await takeScreenshot(component, 'dark-mode-components');
    });
  });
});