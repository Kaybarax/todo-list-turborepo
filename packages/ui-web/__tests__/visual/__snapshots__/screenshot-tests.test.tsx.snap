// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Visual Regression Tests > Badge Component Screenshots > should match badge groups screenshot 1`] = `
<div
  class="p-4 bg-white space-y-4"
>
  <div
    class="flex flex-wrap gap-1"
  >
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        React
      </span>
    </div>
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        TypeScript
      </span>
    </div>
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        Tailwind
      </span>
    </div>
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        Storybook
      </span>
    </div>
  </div>
  <div
    class="flex flex-wrap gap-1"
  >
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        New
      </span>
    </div>
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        Updated
      </span>
    </div>
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        Deprecated
      </span>
    </div>
  </div>
  <div
    class="flex items-center gap-2"
  >
    <span>
      Status:
    </span>
    <div
      class="0 1"
    >
      <span
        class="truncate"
      >
        <span
          class="w-2 h-2 bg-green-500 rounded-full mr-1"
        />
        Active
      </span>
    </div>
  </div>
</div>
`;

exports[`Visual Regression Tests > Badge Component Screenshots > should match badge variants screenshot 1`] = `
<div
  class="p-4 bg-white"
>
  <div
    class="flex flex-wrap gap-2"
  >
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        Default
      </span>
    </div>
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        Secondary
      </span>
    </div>
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        Destructive
      </span>
    </div>
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        Outline
      </span>
    </div>
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        123
      </span>
    </div>
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        <span
          class="w-2 h-2 bg-green-500 rounded-full mr-1"
        />
        Online
      </span>
    </div>
    <div
      class="0"
    >
      <span
        class="truncate"
      >
        üî• Hot
      </span>
    </div>
  </div>
</div>
`;

exports[`Visual Regression Tests > Button Component Screenshots > should match button states screenshot 1`] = `
<div
  class="p-4 space-y-2 bg-white"
>
  <button
    class="0"
  >
    Normal
  </button>
  <button
    class="0"
    disabled=""
  >
    Disabled
  </button>
  <button
    class="0"
    disabled=""
  >
    <svg
      class="mr-2 h-4 w-4 animate-spin"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="opacity-25"
        cx="12"
        cy="12"
        r="10"
        stroke="currentColor"
        stroke-width="4"
      />
      <path
        class="opacity-75"
        d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
        fill="currentColor"
      />
    </svg>
    Loading
  </button>
  <button
    class="0"
    disabled=""
  >
    <svg
      class="mr-2 h-4 w-4 animate-spin"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="opacity-25"
        cx="12"
        cy="12"
        r="10"
        stroke="currentColor"
        stroke-width="4"
      />
      <path
        class="opacity-75"
        d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
        fill="currentColor"
      />
    </svg>
    Please wait...
  </button>
</div>
`;

exports[`Visual Regression Tests > Button Component Screenshots > should match button variants screenshot 1`] = `
<div
  class="p-4 space-y-2 bg-white"
>
  <div
    class="space-x-2"
  >
    <button
      class="0"
    >
      Default
    </button>
    <button
      class="0"
    >
      Destructive
    </button>
    <button
      class="0"
    >
      Outline
    </button>
    <button
      class="0"
    >
      Secondary
    </button>
    <button
      class="0"
    >
      Ghost
    </button>
    <button
      class="0"
    >
      Link
    </button>
  </div>
  <div
    class="space-x-2"
  >
    <button
      class="0"
    >
      Small
    </button>
    <button
      class="0"
    >
      Default
    </button>
    <button
      class="0"
    >
      Large
    </button>
    <button
      class="0"
    >
      üîç
    </button>
  </div>
  <div
    class="space-x-2"
  >
    <button
      class="0"
      disabled=""
    >
      Disabled
    </button>
    <button
      class="0"
      disabled=""
    >
      <svg
        class="mr-2 h-4 w-4 animate-spin"
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          class="opacity-25"
          cx="12"
          cy="12"
          r="10"
          stroke="currentColor"
          stroke-width="4"
        />
        <path
          class="opacity-75"
          d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
          fill="currentColor"
        />
      </svg>
      Loading
    </button>
    <button
      class="0"
    >
      <span
        class="mr-2"
      >
        <span>
          ‚Üê
        </span>
      </span>
      With Icon
    </button>
  </div>
</div>
`;

exports[`Visual Regression Tests > Card Component Screenshots > should match card layouts screenshot 1`] = `
<div
  class="p-4 space-y-4 bg-gray-50"
>
  <div
    class="0"
  >
    <div
      class="0"
    >
      <h3
        class="0"
      >
        Basic Card
      </h3>
      <p
        class="0"
      >
        Simple card layout
      </p>
    </div>
    <div
      class="0"
    >
      <p>
        This is the card content.
      </p>
    </div>
    <div
      class="0"
    >
      <button
        class="0"
      >
        Action
      </button>
    </div>
  </div>
  <div
    class="0"
  >
    <div
      class="0"
    >
      <div
        class="flex items-center justify-between"
      >
        <h3
          class="0"
        >
          Card with Badge
        </h3>
        <div
          class="0"
        >
          <span
            class="truncate"
          >
            New
          </span>
        </div>
      </div>
    </div>
    <div
      class="0"
    >
      <p>
        Card with badge in header.
      </p>
    </div>
  </div>
</div>
`;

exports[`Visual Regression Tests > Card Component Screenshots > should match complex card screenshot 1`] = `
<div
  class="p-4 bg-white"
>
  <div
    class="0"
  >
    <div
      class="0"
    >
      <div
        class="flex items-start justify-between"
      >
        <div>
          <h3
            class="0"
          >
            Project Dashboard
          </h3>
          <p
            class="0"
          >
            Overview of your projects
          </p>
        </div>
        <div
          class="0"
        >
          <span
            class="truncate"
          >
            3 Active
          </span>
        </div>
      </div>
    </div>
    <div
      class="0 1"
    >
      <div
        class="grid grid-cols-2 gap-4"
      >
        <div
          class="text-center"
        >
          <div
            class="text-2xl font-bold"
          >
            12
          </div>
          <div
            class="text-sm text-gray-500"
          >
            Completed
          </div>
        </div>
        <div
          class="text-center"
        >
          <div
            class="text-2xl font-bold"
          >
            3
          </div>
          <div
            class="text-sm text-gray-500"
          >
            In Progress
          </div>
        </div>
      </div>
      <div
        class="space-y-2"
      >
        <div
          class="flex items-center justify-between"
        >
          <span>
            Website Redesign
          </span>
          <div
            class="0"
          >
            <span
              class="truncate"
            >
              Design
            </span>
          </div>
        </div>
        <div
          class="flex items-center justify-between"
        >
          <span>
            Mobile App
          </span>
          <div
            class="0"
          >
            <span
              class="truncate"
            >
              Development
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="0 1"
    >
      <button
        class="0"
      >
        View All
      </button>
      <button
        class="0"
      >
        New Project
      </button>
    </div>
  </div>
</div>
`;

exports[`Visual Regression Tests > Cross-platform Consistency > should match component integration screenshot 1`] = `
<div
  class="p-4 bg-gray-50"
>
  <div
    class="grid grid-cols-1 md:grid-cols-2 gap-4 max-w-4xl"
  >
    <div
      class="0"
    >
      <div
        class="0"
      >
        <div
          class="flex items-center justify-between"
        >
          <h3
            class="0"
          >
            User Profile
          </h3>
          <div
            class="0"
          >
            <span
              class="truncate"
            >
              Active
            </span>
          </div>
        </div>
        <p
          class="0"
        >
          Manage your account settings
        </p>
      </div>
      <div
        class="0 1"
      >
        <div
          class="relative w-full"
        >
          <label
            class="0"
            for=":rc:"
          >
            Display Name
          </label>
          <div
            class="relative"
          >
            <div
              class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"
            >
              <span
                class="0"
              >
                <span>
                  üë§
                </span>
              </span>
            </div>
            <input
              class="0 1"
              id=":rc:"
              value="John Doe"
            />
          </div>
        </div>
        <div
          class="relative w-full"
        >
          <label
            class="0"
            for=":rd:"
          >
            Email
          </label>
          <div
            class="relative"
          >
            <div
              class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"
            >
              <span
                class="0"
              >
                <span>
                  ‚úâÔ∏è
                </span>
              </span>
            </div>
            <input
              class="0 1"
              id=":rd:"
              type="email"
              value="john@example.com"
            />
          </div>
        </div>
      </div>
      <div
        class="0 1"
      >
        <button
          class="0"
        >
          Cancel
        </button>
        <button
          class="0"
        >
          Save Changes
        </button>
      </div>
    </div>
    <div
      class="0"
    >
      <div
        class="0"
      >
        <h3
          class="0"
        >
          Project Status
        </h3>
        <p
          class="0"
        >
          Current project overview
        </p>
      </div>
      <div
        class="0"
      >
        <div
          class="space-y-2"
        >
          <div
            class="flex items-center justify-between"
          >
            <span>
              Website
            </span>
            <div
              class="0"
            >
              <span
                class="truncate"
              >
                In Progress
              </span>
            </div>
          </div>
          <div
            class="flex items-center justify-between"
          >
            <span>
              Mobile App
            </span>
            <div
              class="0"
            >
              <span
                class="truncate"
              >
                Planning
              </span>
            </div>
          </div>
          <div
            class="flex items-center justify-between"
          >
            <span>
              API
            </span>
            <div
              class="0"
            >
              <span
                class="truncate"
              >
                Blocked
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="0"
      >
        <button
          class="0"
        >
          View Details
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Visual Regression Tests > Dark Mode Screenshots > should match dark mode components screenshot 1`] = `
<div
  class="p-4 bg-slate-900 text-white"
>
  <div
    class="space-y-4"
  >
    <div
      class="space-x-2"
    >
      <button
        class="0"
      >
        Default
      </button>
      <button
        class="0"
      >
        Destructive
      </button>
      <button
        class="0"
      >
        Outline
      </button>
    </div>
    <div
      class="0"
    >
      <div
        class="0"
      >
        <h3
          class="0 1"
        >
          Dark Card
        </h3>
        <p
          class="0 1"
        >
          Card in dark theme
        </p>
      </div>
      <div
        class="0"
      >
        <div
          class="relative w-full"
        >
          <div
            class="relative"
          >
            <input
              class="0 3"
              id=":re:"
              placeholder="Dark input..."
            />
          </div>
        </div>
      </div>
      <div
        class="0"
      >
        <button
          class="0"
        >
          Dark Action
        </button>
      </div>
    </div>
    <div
      class="flex gap-2"
    >
      <div
        class="0"
      >
        <span
          class="truncate"
        >
          Dark Badge
        </span>
      </div>
      <div
        class="0"
      >
        <span
          class="truncate"
        >
          Dark Outline
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Visual Regression Tests > Input Component Screenshots > should match input form screenshot 1`] = `
<div
  class="p-4 bg-white"
>
  <form
    class="space-y-4 w-80"
  >
    <div>
      <div
        class="relative w-full"
      >
        <label
          class="0"
          for=":r8:"
        >
          First Name
        </label>
        <div
          class="relative"
        >
          <input
            class="0"
            id=":r8:"
            placeholder="Enter first name..."
            value="John"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        class="relative w-full"
      >
        <label
          class="0"
          for=":r9:"
        >
          Email
        </label>
        <div
          class="relative"
        >
          <div
            class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"
          >
            <span
              class="0"
            >
              <span>
                ‚úâÔ∏è
              </span>
            </span>
          </div>
          <input
            class="0 1"
            id=":r9:"
            placeholder="Enter email..."
            type="email"
            value="john@example.com"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        class="relative w-full"
      >
        <label
          class="0"
          for=":ra:"
        >
          Phone
        </label>
        <div
          class="relative"
        >
          <input
            aria-describedby=":ra:-helper"
            class="0"
            id=":ra:"
            placeholder="Enter phone..."
            type="tel"
          />
        </div>
        <p
          class="0 1"
          id=":ra:-helper"
        >
          Include country code
        </p>
      </div>
    </div>
    <div>
      <div
        class="relative w-full"
      >
        <label
          class="0 1"
          for=":rb:"
        >
          Password
        </label>
        <div
          class="relative"
        >
          <input
            aria-describedby=":rb:-helper"
            class="0"
            id=":rb:"
            placeholder="Enter password..."
            type="password"
          />
        </div>
        <p
          class="0 1"
          id=":rb:-helper"
        >
          Password must be at least 8 characters
        </p>
      </div>
    </div>
  </form>
</div>
`;

exports[`Visual Regression Tests > Input Component Screenshots > should match input variants screenshot 1`] = `
<div
  class="p-4 space-y-4 bg-white w-80"
>
  <div
    class="relative w-full"
  >
    <div
      class="relative"
    >
      <input
        class="0"
        id=":r0:"
        placeholder="Default input"
      />
    </div>
  </div>
  <div
    class="relative w-full"
  >
    <div
      class="relative"
    >
      <input
        class="0"
        id=":r1:"
        placeholder="Email input"
        type="email"
      />
    </div>
  </div>
  <div
    class="relative w-full"
  >
    <div
      class="relative"
    >
      <input
        class="0"
        disabled=""
        id=":r2:"
        placeholder="Disabled input"
      />
    </div>
  </div>
  <div
    class="relative w-full"
  >
    <div
      class="relative"
    >
      <input
        class="0"
        id=":r3:"
        placeholder="Error input"
      />
    </div>
  </div>
  <div
    class="relative w-full"
  >
    <div
      class="relative"
    >
      <div
        class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"
      >
        <span
          class="0"
        >
          <span>
            üîç
          </span>
        </span>
      </div>
      <input
        class="0 1"
        id=":r4:"
        placeholder="With left icon"
      />
    </div>
  </div>
  <div
    class="relative w-full"
  >
    <div
      class="relative"
    >
      <input
        class="0 2"
        id=":r5:"
        placeholder="With right icon"
      />
      <div
        class="absolute inset-y-0 right-0 flex items-center pr-3 pointer-events-none"
      >
        <span
          class="0"
        >
          <span>
            ‚úì
          </span>
        </span>
      </div>
    </div>
  </div>
  <div>
    <div
      class="relative w-full"
    >
      <label
        class="0"
        for=":r6:"
      >
        Complete Input
      </label>
      <div
        class="relative"
      >
        <input
          aria-describedby=":r6:-helper"
          class="0"
          id=":r6:"
          placeholder="With label and helper"
        />
      </div>
      <p
        class="0 1"
        id=":r6:-helper"
      >
        This is helper text
      </p>
    </div>
  </div>
  <div>
    <div
      class="relative w-full"
    >
      <label
        class="0 1"
        for=":r7:"
      >
        Error Input
      </label>
      <div
        class="relative"
      >
        <input
          aria-describedby=":r7:-helper"
          class="0"
          id=":r7:"
          placeholder="With error state"
        />
      </div>
      <p
        class="0 1"
        id=":r7:-helper"
      >
        This is an error message
      </p>
    </div>
  </div>
</div>
`;
